snet_cli/commands.py

```diff
@@ -813,6 +812,102 @@ def _getserviceregistrationbyname(self, service_json=None):
+                    os.makedirs(codegen_tmp_dir)
+                proto_include = pkg_resources.resource_filename("grpc_tools", "_proto")
+
+                # Checking if protobuf files are valid by trying to compile them
+                for idx, proto_path in enumerate(import_paths):
+                    protoc_args = [
+                        "protoc",
+                        "-I{}".format(entry_path),
+                        '-I{}'.format(proto_include),
+                        "--python_out={}".format(codegen_tmp_dir),
+                        "--grpc_python_out={}".format(codegen_tmp_dir),
+                        str(proto_path)
+                    ]
+                    # stdout suppression
+                    sys.stdout = open(os.devnull, 'w')
+                    if not protoc(protoc_args):
+                        valid_import_paths.add(proto_path)
+                    else:
+                        sys.stdout = sys.__stdout__
+                        self._printerr("{} is not a valid protobuf file.\n".format(proto_path.name))
+
+                # stdout back to console
+                sys.stdout = sys.__stdout__
+                if os.path.exists(codegen_tmp_dir):
+                    rmtree(codegen_tmp_dir)
+
+        # Create service_spec/ tar and upload it to IPFS
+        ipfs_endpoint = urlparse(self.config["ipfs"]["default_ipfs_endpoint"])
+        ipfs_scheme = ipfs_endpoint.scheme if ipfs_endpoint.scheme else "http"
+        ipfs_port = ipfs_endpoint.port if ipfs_endpoint.port else 5001
+        ipfs_client = ipfsapi.connect(urljoin(ipfs_scheme, ipfs_endpoint.hostname), ipfs_port)
+        spec_ipfs_uri = None
+
+        if valid_import_paths:
+            tmp_f = self._tar_imports(valid_import_paths, entry_path)
+            spec_ipfs_hash = ipfs_client.add(read_temp_tar(tmp_f).name)["Hash"]
+            spec_ipfs_path = "/ipfs/{}".format(spec_ipfs_hash)
+            spec_ipfs_uri = uri_reference(spec_ipfs_path).copy_with(scheme='ipfs').unsplit()
+
+        # Upload metadata JSON to IPFS with modelURI
+        metadata_json = dict(service_json['metadata'])
+
+        if spec_ipfs_uri:
+            metadata_json['modelURI'] = spec_ipfs_uri
+
+        with tempfile.NamedTemporaryFile(mode='w+') as tmp_json:
+            json.dump(metadata_json, tmp_json, ensure_ascii=False, sort_keys=True)
+            tmp_json.seek(0)
+            metadata_ipfs_hash = ipfs_client.add(tmp_json.name)["Hash"]
+        metadata_ipfs_path = "/ipfs/{}".format(metadata_ipfs_hash)
+        metadata_ipfs_uri = uri_reference(metadata_ipfs_path).copy_with(scheme='ipfs').unsplit()
+
+        # Checking price
+        if not type(service_json["price"]) == int:
+            self._printerr("Invalid price format: {} [{}]".format(service_json["price"], type(service_json["price"])))
+            is_ok = False
+
+        # Checking endpoint
+        try:
+            request = requests.get(str(service_json["endpoint"]), timeout=5)
```

Comment:
Do not think we should actually check that endpoint is live as it prevents us from publishing service which is temporary down. Is it enough to check that endpoint has correct URI format?

Reply:
Even if the endpoint is unreachable the user can type "y" and proceed with the publishing/update.
If we check it with something like `urlparse` (is this what you're suggesting?) some mistyping can pass unnoticed, like: http://singulritynet.io:7000.

Reply:
Ah ok didn't catch it at first time. If we can proceed even if service is down that is ok I think. Thanks!